cards4-template.js

```diff
@@ -3,10 +3,39 @@ var makeCard = // receive factory with external name `makeCard`
 
     // The factory itself:
     function makeCard(id) {  //makeCard is also IIFE's internal name
-        // set instance properties here
+      if (makeCard.isValidID(id)){
+        return {
+
+          id: id,
+          rank: rank,
+          suit: suit,
+          color: color,
+          name: cardName
+
+          }
+        } else{
+          return null;
+     }
+    };
+
+
+// //-----------------------
+// // Methods to be called through factory only:
+// //-----------------------
+makeCard.isValidID = function(id){
+    return (typeof id === "number")&&(id%1===0)&&(id>=0)&&(id<=51);
+};
+
+makeCard.isCard = function(thing) {
+        return (typeof thing === 'object')&&(makeCard.rank === thing.rank)&&('id' in thing)&&(makeCard.isValidID(thing.id));
+};    // set instance properties here
+
+var rankNames = ['','Ace','Two','Three','Four','Five','Six','Seven','Eight','Nine','Ten','Jack','Queen','King'],
```

Comment:
Slight formatting error on this line (33, and again on line 35), replace your 's at the end of the arrays with ;'s.